### PR TITLE
Hydrostatics for RAO

### LINKED ISSUE #64: automated hydrostatics computation
To get hydrostatics when using `capytaine` (and Nemoh), I mostly use [`Meshmagick`](https://github.com/LHEEA/meshmagick). This goes something like 

```python
from meshmagick import mesh as mm
from meshmagick import hydrostatics as hs

fb = cpt.FloatingBody.from_file(filename=fbname+'.stl', name=fbname)
hsa = hs.Hydrostatics(mm.Mesh(fb.mesh.vertices, fb.mesh.faces))
hsd = hsa.hs_data
m = hsd['disp_mass']

I = np.array([[hsd['Ixx'], -1*hsd['Ixy'], -1*hsd['Ixz']],
              [-1*hsd['Ixy'], hsd['Iyy'], -1*hsd['Iyz']],
              [-1*hsd['Ixz'], -1*hsd['Iyz'], hsd['Izz']]])
M = block_diag(m, m, m, I) # TODO - scale inertias if mass is specified by user

fb.mass = self.fb.add_dofs_labels_to_matrix(M)
kHS = block_diag(0,0,hsd['stiffness_matrix'],0)

fb.hydrostatic_stiffness = self.fb.add_dofs_labels_to_matrix(kHS)
```

@mancellin - what are your thoughts about incorporating this into `capytaine`?

### PR DESCRIPTION
**Pull request by ryancoe**
There are a number of discussions/requests about computing hydrostatics (#64, #36, #55, #66). However, as noted in #64 including automated calculation of hydrostatics in `capytaine` is not so easy, as it support generalized modes. This PR updates the [`boat_animation`](https://github.com/mancellin/capytaine/blob/87d09bf56dcd1d4a242a29805e9d49511b3947e1/docs/user_manual/examples/boat_animation.py) example to show how the hydrostatics can be computed using [`meshmagick`](https://github.com/LHEEA/meshmagick) if desired.

closes #64 #66

### CODE REVIEW

**Comment by ryancoe** [comment]
I was considering adding the inertia/mass computation in here (assuming neutrally buoyant and uniformly distributed body), but I noticed that the mass matrix currently being used 

```python
boat.mass = boat.add_dofs_labels_to_matrix(
        [[1e4, 0,   0,   0,   0,   0],
         [0,   1e4, 0,   0,   0,   0],
         [0,   0,   1e4, 0,   0,   0],
         [0,   0,   0,   1e7, 0,   2e5],
         [0,   0,   0,   0,   4e7, 0],
         [0,   0,   0,   2e5, 0,   5e7]]
    )
```

matches `meshmagick` for the moments of inertia, but not the rigid body mass (i.e., `1e4 != 9.552e+05`)

```python
hsd = hs.Hydrostatics(mm.Mesh(boat.mesh.vertices, boat.mesh.faces)).hs_data
m = hsd['disp_mass']
kHS = block_diag(0,0,hsd['stiffness_matrix'],0)
boat.hydrostatic_stiffness = boat.add_dofs_labels_to_matrix(kHS)

assert hsd['disp_volume']*1023 == hsd['disp_mass'] # checking the units...

I = np.array([[hsd['Ixx'], -1*hsd['Ixy'], -1*hsd['Ixz']],
              [-1*hsd['Ixy'], hsd['Iyy'], -1*hsd['Iyz']],
              [-1*hsd['Ixz'], -1*hsd['Iyz'], hsd['Izz']]])
M = block_diag(m, m, m, I)
print(M)
```
```
array([[9.552e+05, 0.000e+00, 0.000e+00, 0.000e+00, 0.000e+00, 0.000e+00],
       [0.000e+00, 9.552e+05, 0.000e+00, 0.000e+00, 0.000e+00, 0.000e+00],
       [0.000e+00, 0.000e+00, 9.552e+05, 0.000e+00, 0.000e+00, 0.000e+00],
       [0.000e+00, 0.000e+00, 0.000e+00, 1.019e+07, 5.774e-02, 2.004e+05],
       [0.000e+00, 0.000e+00, 0.000e+00, 5.774e-02, 4.139e+07, 2.209e-02],
       [0.000e+00, 0.000e+00, 0.000e+00, 2.004e+05, 2.209e-02, 4.919e+07]])
```

**Comment by mancellin** [comment]
I agree on adding this kind of code sample to the documentation.
Maybe, creating a new independent recipe "Computing the hydrostatics with meshmagick" will make it easier to find (for instance in the table of content of the cookbook).

Not sure what happened with the mass, probably a mistake...

**Comment by ryancoe** [comment]
- [ ] **example dedicated to hydrostatics and mass properties:** 6d8a7798446e4dcc00dda5f7dc0c104cd5c33383
- [ ] **correct hard-coded mass in [`boat_animation.py`](https://github.com/mancellin/capytaine/blob/3146508147f21d8e7ffab0f787e0849996333bf0/docs/user_manual/examples/boat_animation.py):** 4e8cc954f4b76d7286bb29fabcc314aa38a640a4

**Comment by sharmi0** [comment]
This is helpful for the hydrostatics, but is there a way to calculate the inertia moments by using meshmagick? Thank you!

File: docs/user_manual/examples/boat_animation.py [hunk 1]
```diff
@@ -7,6 +7,16 @@
 import capytaine as cpt
 from capytaine.ui.vtk import Animation
 
+from capytaine.tools.optional_imports import import_optional_dependency
```
Comment (mancellin) [inline docs/user_manual/examples/boat_animation.py:10]: Are you actually using that?

**Comment by ryancoe** [comment]
@sharmi0 - We added an example specifically for doing this: [`hydrostatics.py`](https://github.com/mancellin/capytaine/blob/ac0de8412b0e4cde8f8dd3ac7c12818308743f16/docs/user_manual/examples/hydrostatics.py). I can also add this to the RAO example I suppose depending on @mancellin's preference?

File: docs/user_manual/examples/boat_animation.py (hunk 1 above)
Comment (ryancoe) [reply docs/user_manual/examples/boat_animation.py]: no! fixed with b829008

**Comment by mancellin** [comment]
Looks good to me.

**Pull request merged**